dbt/adapters/athena/lf_permissions.py

```diff
@@ -0,0 +1,144 @@
+from functools import cached_property
+from typing import Dict, List
+
+from mypy_boto3_lakeformation import LakeFormationClient
+from mypy_boto3_lakeformation.type_defs import (
+    BatchPermissionsRequestEntryTypeDef,
+    DataCellsFilterTypeDef,
+)
+from mypy_boto3_sts import STSClient
+from pydantic import BaseModel
+
+from dbt.adapters.athena.relation import AthenaRelation
+from dbt.events import AdapterLogger
+
+logger = AdapterLogger("AthenaLakeFormation")
+
+
+class FilterConfig(BaseModel):
+    row_filter: str
+    column_names: List[str] = []
+    principals: List[str] = []
+
+    def to_api_repr(self, catalog_id: str, database: str, table: str, name: str) -> DataCellsFilterTypeDef:
+        return {
+            "TableCatalogId": catalog_id,
+            "DatabaseName": database,
+            "TableName": table,
+            "Name": name,
+            "RowFilter": {"FilterExpression": self.row_filter},
+            "ColumnNames": self.column_names,
+            "ColumnWildcard": {"ExcludedColumnNames": []},
+        }
+
+    def to_update(self, existing: DataCellsFilterTypeDef) -> bool:
+        return self.row_filter != existing["RowFilter"]["FilterExpression"] or set(self.column_names) != set(
+            existing["ColumnNames"]
+        )
+
+
+class DataCellFiltersConfig(BaseModel):
+    enabled: bool = False
+    filters: Dict[str, FilterConfig]
+
+
+class LfGrantsConfig(BaseModel):
+    data_cell_filters: DataCellFiltersConfig
+
+
+class LfPermissions:
+    def __init__(self, relation: AthenaRelation, lf: LakeFormationClient, sts: STSClient) -> None:
+        self.relation = relation
+        self.database: str = relation.schema
+        self.table: str = relation.identifier
+        self.lf = lf
+        self.sts = sts
```

Comment:
Why using `sts` to get the `calatog_id` and not retrieving it with glue API as we do at other places in the code ?

Reply:
I get the following error when I try use default athena client call: 
```
An error occurred (InvalidRequestException) when calling the GetDataCatalog operation: DataCatalog AwsDataCatalog was not found
```

that's why it seems to be the most optimal way to get AWS Account Id